### PR TITLE
(PCP-530) Allow cert chains in client cert

### PR DESCRIPTION
**Pull request by MikaelSmith**
Allow chained certificates in the client cert. Previously an error would
be generated if the cert file contained more than one certificate.

### CODE REVIEW

File: src/puppetlabs/pcp/client.clj [hunk 1]
```diff
@@ -142,10 +142,13 @@
...
   [certificate type]
-  (let [x509     (ssl-utils/pem->cert certificate)
-        cn       (ssl-utils/get-cn-from-x509-certificate x509)
-        identity (format "pcp://%s/%s" cn type)]
-    identity))
+  (let [x509-chain (ssl-utils/pem->certs certificate)]
+    (when (empty? x509-chain)
+      (throw (IllegalArgumentException.
+               (i18n/trs "{0} must contain at least 1 certificate" certificate))))
+    (let [cn       (ssl-utils/get-cn-from-x509-certificate (first x509-chain))
```
Comment (parisiale) [inline src/puppetlabs/pcp/client.clj]: What guarantees that the client cert will be the first one? Sorry, but I'm not familiar with certificate bundles in action.

Comment (parisiale) [inline src/puppetlabs/pcp/client.clj]: What do you say about adding a couple of lines in the `connect` docstring (last function, on the bottom) about our assumption on the certs bundle?

**Comment by parisiale** [comment]
👍  
Just a couple of questions.

File: src/puppetlabs/pcp/client.clj (hunk 1 above)
Comment (MikaelSmith) [inline src/puppetlabs/pcp/client.clj]: I've found several sources saying it's ordered, starting with the leaf certificate first - https://support.dnsimple.com/articles/what-is-ssl-certificate-chain/, http://superuser.com/a/347614 - but nothing authoritative.

**Comment by MikaelSmith** [comment]
Updated.

**Pull request merged**